src/python/pants/goal/run_tracker.py

```diff
@@ -387,15 +390,51 @@ def store_stats(self):
     if target_data:
       run_information['target_data'] = ast.literal_eval(target_data)
 
+    def get_option(scope_or_scope_with_flag, flag=None):
+      """Looks up a scope in the options parsed by Pants.
+
+      scope_or_scope_with_flag is a string either either of form scope or scope.flag
+      If it's a scope, the scope's values will be gotten as a dict, and either returned if flag is
+        None, or else the flag will be looked up inside the dict.
+      If it's a scope.flag, the scope will be looked up and the flag looked up and returned.
+      If the scope or flag can't be found, ValueError will be raised.
+      """
+      scope_to_look_up = scope_or_scope_with_flag if scope_or_scope_with_flag != 'GLOBAL' else ''
+      try:
+        # Yeah, it's a little sketchy to be looking up Subsystem._options here, but... Someone went
+        # to some care to make sure that Subsystems only get scoped options, and we need to
+        # circumvent that in this one specific place...
+        value = Subsystem._options.for_scope(scope_to_look_up, inherit_from_enclosing_scope=False).as_dict()
+        if flag is None:
+          return value
+        else:
+          return value[flag]
+      except Config.ConfigValidationError:
+        # We only support one level of scope.flag - if we already were looking up a flag, or there
+        # are no parent scopes, don't traverse into parent scopes.
+        if flag is None and '.' in scope_to_look_up:
+          # Option wasn't a scope - maybe it was a scope.flagname?
+          return get_option(*scope_to_look_up.rsplit('.', 1))
+        raise ValueError("Couldn't find option scope {} for recording".format(scope))
+      except AttributeError:
+        # If we looked up a flag in a set of scopes, and it wasn't found, error.
+        raise ValueError("Couldn't find option scope {} for recording".format(scope))
+
+    recorded_options = {}
+    for scope in self.get_options().stats_option_scopes_to_record:
```

Comment:
I'm not sure `store_stats` is the right place for computing these values. All the other stats are computed elsewhere and this method just pulls them together.  I would put all this logic into a separate method, and call that method in `end()`.

Reply:
Moved into a separate method. Still calling it here, not in `end()`, because introducing a field just to communicate between `end` and `store_stats` seems weird.